src/code/z_kankyo.c

```diff
@@ -832,9 +2728,267 @@ u8 sSandstormEnvColors[] = {
     50,  40,  0,   //
 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_DrawSandstorm.s")
+void Environment_DrawSandstorm(PlayState* play, u8 sandstormState) {
+    s32 primA1 = 0;
+    s32 envA1 = 0;
+    s32 primA = play->envCtx.sandstormPrimA;
+    s32 envA = play->envCtx.sandstormEnvA;
+    s32 step = 9;
+    u16 sp96;
+    u16 sp94;
+    u16 sp92;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+    s32 index;
+    f32 sp98;
+    s32 pad;
+
+    switch (sandstormState) {
+        case SANDSTORM_ACTIVE:
+            primA1 = play->state.frames % 128;
+            if (primA1 > 128) {
+                primA1 = 255 - primA1;
+            }
+            primA1 += 73;
+            envA1 = 255;
+            break;
+
+        case SANDSTORM_FILL:
+            primA1 = 255;
+            envA1 = (play->envCtx.sandstormPrimA >= 255) ? 255 : 128;
+            break;
+
+        case SANDSTORM_UNFILL:
+        case SANDSTORM_5:
+            envA1 = 128;
+            if (play->envCtx.sandstormEnvA > 128) {
+                primA1 = 255;
+            } else {
+                primA1 = play->state.frames % 128;
+                if (primA1 > 64) {
+                    primA1 = 128 - primA1;
+                }
+                primA1 += 73;
+            }
+            if ((primA1 >= primA) && (primA1 != 255)) {
+                play->envCtx.sandstormState++;
+            }
+            break;
+
+        case SANDSTORM_DISSIPATE:
+            primA1 = (play->envCtx.sandstormEnvA > 128) ? 255 : play->envCtx.sandstormEnvA >> 1;
+
+            if (primA == 0) {
+                play->envCtx.sandstormState = SANDSTORM_OFF;
+            }
+            break;
+
+        case SANDSTORM_6:
+        case SANDSTORM_8:
+            primA1 = D_801F4E30;
+            envA1 = D_801F4E30;
+            step = 1;
+            if (sandstormState == SANDSTORM_8) {
+                step = 0xA;
```

Comment:
```suggestion
                step = 10;
```
step should probably be decimal (looks like a color)